File: treatment/src/main/java/com/hartwig/actin/treatment/ctc/CTCStatus.java
package com.hartwig.actin.treatment.ctc;

import java.util.Set;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.jetbrains.annotations.NotNull;

public enum CTCStatus {
    OPEN,
    CLOSED;

    private static final Logger LOGGER = LogManager.getLogger(CTCStatus.class);

    private static final Set<String> OPEN_STATES = Set.of("Open");
    private static final Set<String> CLOSED_STATES = Set.of("Gesloten", "Nog niet geopend", "Onbekend", "Tijdelijk gesloten");

    @NotNull
    public static CTCStatus fromStatusString(@NotNull String string) {
        if (OPEN_STATES.stream().anyMatch(state -> state.equalsIgnoreCase(string))) {
            return OPEN;
        } else if (CLOSED_STATES.stream().anyMatch(state -> state.equalsIgnoreCase(string))) {
            return CLOSED;
        }

        LOGGER.warn("Could not interpret CTC status string: '{}'. Assuming status implies {}}", string, CLOSED.toString());
        return CLOSED;
    }
}

File: treatment/src/main/kotlin/com/hartwig/actin/treatment/ctc/CTCStatus.kt
package com.hartwig.actin.treatment.ctc

import org.apache.logging.log4j.LogManager

internal enum class CTCStatus {
    OPEN, CLOSED;

    companion object {
        private val LOGGER = LogManager.getLogger(CTCStatus::class.java)
        private val OPEN_STATES = setOf("Open")
        private val CLOSED_STATES = setOf("Gesloten", "Nog niet geopend", "Gesloten voor inclusie", "Onbekend", "Tijdelijk gesloten")

        fun fromStatusString(string: String): CTCStatus {
            return when {
                OPEN_STATES.any { it.equals(string, ignoreCase = true) } -> OPEN

                CLOSED_STATES.any { it.equals(string, ignoreCase = true) } -> CLOSED

                else -> {
                    LOGGER.warn(" Could not interpret CTC status string: '{}'. Assuming status implies {}}", string, CLOSED.toString())
                    CLOSED
                }
            }
        }
    }
}